### PR TITLE
Fix: getCanGroup always resolving to true

### PR DESCRIPTION
**Pull request by mozzerbob**
Small update to getCanGroup. It would always resolve to true and ignore the rest of the statement.

Fixing issue #4809.

### CODE REVIEW

**Comment by step2341** [comment]
@mozzerbob i can see the fix above, im using "@tanstack/react-table": "^8.10.3", and still having this issue. has this fix been released?

**Comment by mozzerbob** [comment]
> @mozzerbob i can see the fix above, im using "@tanstack/react-table": "^8.10.3", and still having this issue. has this fix been released? @KevinVandy i noticed you have this option in your materialTable, how did you get it to work correctly?

It never got merged, its waiting for a maintainer approval.

**Comment by tannera-cargill** [comment]
Is it possible to get this merged? This function always returns true, even when enableGrouping is false.

**Review by KevinVandy — APPROVED** [review]

**Pull request merged**
